### ISSUE TITLE
Can't install wheels with non-normalized names

### ISSUE THREAD

**Issue by stefanor** [pypa/installer#134]
Names are parsed from the distribution filename, but never normalized.

If a distribution has a non-normalized filename, like `Quart-0.17.0-py3-none-any.whl`, then installation fails:

```Traceback (most recent call last):
  File "/usr/bin/pybuild", line 367, in main
    run(func, i, version, c)
  File "/usr/bin/pybuild", line 317, in run
    result = func(context, args)
  File "/usr/share/dh-python/dhpython/build/plugin_pyproject.py", line 102, in build
    self.build_step2(context, args)
  File "/usr/share/dh-python/dhpython/build/plugin_pyproject.py", line 151, in build_step2
    install(
  File "/usr/lib/python3/dist-packages/installer/_core.py", line 77, in install
    root_scheme = _process_WHEEL_file(source)
  File "/usr/lib/python3/dist-packages/installer/_core.py", line 21, in _process_WHEEL_file
    stream = source.read_dist_info("WHEEL")
  File "/usr/lib/python3/dist-packages/installer/sources.py", line 139, in read_dist_info
    return self._zipfile.read(path).decode("utf-8")
  File "/usr/lib/python3.10/zipfile.py", line 1475, in read
    with self.open(name, "r", pwd) as fp:
  File "/usr/lib/python3.10/zipfile.py", line 1514, in open
    zinfo = self.getinfo(name)
  File "/usr/lib/python3.10/zipfile.py", line 1441, in getinfo
    raise KeyError(
KeyError: "There is no item named 'Quart-0.17.0.dist-info/WHEEL' in the archive"
```

From: https://bugs.debian.org/1008606
Part of the bigger issue in #97.

**Comment by dimaqq** [comment]
I've played a bit with `poetry==1.4.0` and I'm pretty sure it uses `installer==0.6.0`, the latest released.

Yet the above, linked error still happens.

It seems that the fix was not released?

**Comment by baurt** [comment]
This was closed about a week after the `0.6.0` release was cut, so it would make sense that it doesn't work.

> I've played a bit with `poetry==1.4.0` and I'm pretty sure it uses `installer==0.6.0`, the latest released.
> 
> Yet the above, linked error still happens.
> 
> It seems that the fix was not released?

This was closed about a week after the `0.6.0` release was cut, so it would make sense that it doesn't work.
